### ISSUE TITLE
[3.6.6] Issues related by moveing cinnamon theme into numix frost repo

### ISSUE THREAD

**Issue by jeweloper** [Antergos/Numix-Frost#48]
Since moving cinnamon theme into numix-frost-repo (under v3.6.6), there are some issues with the cinnamon theme.
By now I notice the following issues:
- panel tooltip has full transparent bg (expected: opaque dark)
- OSD panel has full transparent bg (expected: opaque dark)
- desktop panel has semitransparent dark bg (expected: opaque dark)
- (main)menu and calendar applet with black fonts on dark bg (expected: a light font)
- (main)menu controls missing its themes (e.g. scroll-bar, search-box)
- no space between popup-item-icon and popup-item-text
- applet/popup-item font size increased

@lots0logs I will help you to fix these issues. Please reply to me, if you intended some of these chances will be a permanent replacement. Or if there is something else I should know.
By now I will use v3.6.5 as reference.

**Comment by lots0logs** [comment]
@jeweloper I had not planned on releasing the cinnamon theme changes until it was actually ready but there was a packaging issue that came up which forced my hand (an update to a core package in Arch broke cinnamon for antergos users) so I had to do the release. 

I think I mentioned my plans already in one of your other PRs? Maybe not, I am not sure. Anyway, I want to replace the usages of svg images by pure CSS where possible. However, considering that there are items which are broken at the moment I guess its a good idea to fix those (with the svgs) so that they work until we have time to actually write CSS to replace the svgs.

**Comment by jeweloper** [comment]
@lots0logs Yes, that you want to replace svg by pure css you mentioned already.
One specific question: In ⋜3.6.5 the main bg-color was hard coded with opaque dark (#444). What becomes the new bg-color replacement for window-titlebar, OSD, tooltip, etc?
$panel_bg_color, $menu_bg_color, $dark_bg_color or something else.
Thx!

**Comment by lots0logs** [comment]
@jeweloper Check _global.scss (all colors are defined there). I know I deviated a bit from them with the panel though :sweat_smile:

**Comment by jeweloper** [comment]
@lots0logs By now, I have had only time to look at  [_calendar.scss](https://github.com/Antergos/Numix-Frost/blob/master/src/desktops/cinnamon/scss/applets/_calendar.scss). And there are some issues I couldn't fix. But maybe you've got some answers for me.
1. In the first step I've tried to replace the arrow-images by [pure css](https://css-tricks.com/snippets/css/css-triangle/), but I had no success. Secondly I've I created I resource-file with all the required images, but also without success. (Is it even possible to reference to a gresource file within cinnamon.css or is it only a gtk feature?)
So, is it ok, if I continue to reference all required images to the sub-directories "widget-assets", "panel-assets", etc.?
2. Some colors in the compiled cinnamon.css are set with the non compiled sass-function mix(). Is it because of the [wrong format](http://sass-lang.com/documentation/Sass/Script/Functions.html#mix-instance_method) of the percent parameter (percent value instead of ratio)? For example: `mix(#eee,#444,20%)`

Thx, for your answers.

**Comment by lots0logs** [comment]
@jeweloper I reworked the assets so now all of cinnamons assets will be accessible in the generated css file at this path: `../assets/cinnamon/some-asset-file.svg` For things that cannot be easily accomplished in CSS we can continue using the svg assets. The main thing is to get rid of any backgrounds and border images that use svgs. 

I'm not sure about the reasoning behind how mix is implemented in _funcions.scss. I'll have to look into it. What exactly is the issue, I mean is something not correct in the generated CSS?

**Comment by jeweloper** [comment]
@lots0logs I've no skills in css nor in sass, but I think the mix-function should compiled into a pure hex-value. Right? It seems this is not the case.
e.g. the color of calendar-other-month-day is defined by `$osd_insensitive_fg_color: mix($osd_fg, $osd_base, .6)`. And it is "only" compiled to `mix(#eee,#444,0.6);`. I supposed it is because the percentage is specified by its ratio and not by its percent value (followed by percent symbol) like the doc said.
In short: `$osd_insensitive_fg_color: mix($osd_fg, $osd_base, 60%)` should compiled to a hex-value. But maybe I'm wrong and ratio is also a valid format and there is another reason that it is not fully compiled.

**Comment by lots0logs** [comment]
@jeweloper I think that the intention is to let GTK mix the colors. I don't know why that decision was made but I will try to find out. Though now that I've typed that it occurs to me that Cinnamon probably doesn't offer the full set of color functions that are available in GTK themes :doh: I'll come up with a work-around asap.

**Comment by jeweloper** [comment]
@lots0logs Thx for your effort.

Today I've tried to fix u.a. the .menu/.popup-menu and the WindowList. But here I'm in conflict with your last commits.

In opposite to your [.menu/.popup-menu redesign](https://github.com/Antergos/Numix-Frost/blob/master/src/desktops/cinnamon/scss/components/_menu.scss) I used the current gtk3 context menu as my sample. (Which basically corresponds with the known v3.6.5. design.):
- only a light shadow and light roundness
- small item spacing
- no right/left border, small top/bottom border
What are your plans? Should I first adapt your design pattern, before I open my pull request? Or is your redesign only a kind of a concept? In either case I will integrate your darken [separator-menu-item](https://github.com/Antergos/Numix-Frost/blob/master/src/desktops/cinnamon/scss/components/_menu.scss#L76) and [sub-menu](https://github.com/Antergos/Numix-Frost/blob/master/src/desktops/cinnamon/scss/components/_menu.scss#L58). This looks pretty cool.

I think your changes in [WindowList](https://github.com/Antergos/Numix-Frost/blob/master/src/desktops/cinnamon/scss/applets/_window-list.scss) attempts to fix the issue with these truncated tooltips. Right?
But I think this has to be fixed in the applet itself. There is already an [open issue](https://github.com/jaszhix/icingtaskmanager/issues/39), so I think this will be fixed very soon.
So that's why my pull request will revoke your changes.

**Comment by lots0logs** [comment]
@jeweloper While I'm still not 100% happy with the popup-menus, I do consider them to be good enough for now. I don't want them to look exactly like the GTK menus because the desktop shell should be easily distinguishable from GTK (IMO). I have not had a chance to get to the menu applet, but it is next on my todo list. Ultimately the goal is to have a style that blends well and feels consistent with the GTK theme (but not necessarily identical to it in all areas). Does that make sense?

**Comment by jeweloper** [comment]
@lots0logs Yes, I think so.
Personally I would favor a design that's consistent in its shell, gtk3 and gtk2 design.
Maybe it is possible to switch between the old and new style only by editing the configuration file (_global.scss).
I think if you go on and expand the `// widget styles ` params this could be possible.
With these expansion you could also switch easily to the new style later on, when the changes also implemented in gtk3 and gtk2 theme and everything is consistence.
And it should also be possible to easily compile a special Theme - optimized for e.g. touch displays (large space between menu items, etc.).
(And yes, guys like me can easily compile there own version with the old look and feel. :stuck_out_tongue_winking_eye:)

But I don't know if it's worth the trouble.
If you like, I can have a look.

**Comment by jeweloper** [comment]
@lots0logs A administrative question: Is it helpful if I already open a pull request, even if I'm still in progress? Or is it well if I open a separate issue (e.g. "fixing calendar applet") and link it with a finalized pull request?
Thx.

**Comment by lots0logs** [comment]
>Personally I would favor a design that's consistent in its shell, gtk3 and gtk2 design.

Yes, of course. I agree. Though consistency doesn't always mean identical. For example, looking at the popup menus compared to the one's in GTK..there isnt very much difference:

![2016-11-17_04-35-38-pm2](https://cloud.githubusercontent.com/assets/4675662/20612538/adcf6c6e-b280-11e6-9b11-06002ab93ee8.png) ![2016-11-24_07-47-46-pm](https://cloud.githubusercontent.com/assets/4675662/20612540/b47f5768-b280-11e6-86a4-ae6b4dbebcc9.png)

Is it the background color that you don't like? 

> Is it helpful if I already open a pull request, even if I'm still in progress?

Sure. You can just note that its a WIP and once its complete you can ping me.

**Comment by jeweloper** [comment]
Looks perfect!
Sorry, I think I referred to an outdated version or a faulty and inconsistent self compiled version :confused:.
Forget what I wrote.
It's time to immediately tidy up my project files. :see_no_evil:

**Comment by devland** [comment]
I don't know if this will help, but there are some numix desktop theme variations online that still look ok.

The "numix cinnamon transparent" one appears to be completely unaltered. Maybe you can reverse engineer it to fix numix-frost. :)

**Comment by ricflomag** [comment]
While it's being solved, why not changing cinnamon default desktop theme setting from Numix-Frost to Numix-SX-Dark (not installed by default right now, but seems flawless) ?

Fresh installs would look better until it is solved :)

**Comment by ricflomag** [comment]
Simplier : set it to "Linux-Mint", which is installed by default.
That's antergos-cinnamon-setup package, and /usr/share/glib-2.0/schemas/90_antergos_cinnamon.gschema.override should read (line : 18 and 19)

```
[org.cinnamon.theme]
name='Linux-Mint'
```
instead of
```
[org.cinnamon.theme]
name='Numix-Frost'
```

Who can do that ?
